executor/show.go

```diff
@@ -386,6 +386,11 @@ func (e *ShowExec) fetchShowColumns() error {
 				columnDefault = defaultValStr
 			}
 		}
+		switch col.Tp {
+		case mysql.TypeTimestamp, mysql.TypeDate, mysql.TypeDuration, mysql.TypeDatetime,
+			mysql.TypeYear, mysql.TypeNewDate:
+			desc.Collation = "NULL"
```

Comment:
Do we need to check other types like `int`?
```
mysql> create table t1 (a date,b datetime,c timestamp, d int);
Query OK, 0 rows affected (0.02 sec)

mysql> show full columns from t1;
+-------+-----------+-----------+------+-----+-------------------+-----------------------------+---------------------------------+---------+
| Field | Type      | Collation | Null | Key | Default           | Extra                       | Privileges                      | Comment |
+-------+-----------+-----------+------+-----+-------------------+-----------------------------+---------------------------------+---------+
| a     | date      | NULL      | YES  |     | NULL              |                             | select,insert,update,references |         |
| b     | datetime  | NULL      | YES  |     | NULL              |                             | select,insert,update,references |         |
| c     | timestamp | NULL      | NO   |     | CURRENT_TIMESTAMP | on update CURRENT_TIMESTAMP | select,insert,update,references |         |
| d     | int(11)   | NULL      | YES  |     | NULL              |                             | select,insert,update,references |         |
+-------+-----------+-----------+------+-----+-------------------+-----------------------------+---------------------------------+---------+
4 rows in set (0.01 sec)
mysql> create table t1 (a date,b datetime,c timestamp, d int);
Query OK, 0 rows affected (0.00 sec)

mysql> show full columns from t1;
+-------+-----------+-----------+------+------+---------+-------+---------------------------------+---------+
| Field | Type      | Collation | Null | Key  | Default | Extra | Privileges                      | Comment |
+-------+-----------+-----------+------+------+---------+-------+---------------------------------+---------+
| a     | date      | binary    | YES  |      | NULL    |       | select,insert,update,references |         |
| b     | datetime  | binary    | YES  |      | NULL    |       | select,insert,update,references |         |
| c     | timestamp | binary    | YES  |      | NULL    |       | select,insert,update,references |         |
| d     | int(11)   | binary    | YES  |      | NULL    |       | select,insert,update,references |         |
+-------+-----------+-----------+------+------+---------+-------+---------------------------------+---------+
4 rows in set (0.00 sec)
```

Reply:
I think it should be filed in another issue.

Reply:
https://github.com/pingcap/tidb/issues/9837
Filed the issue to trace this problem.